src/components/content/order/create/OrderSubmitStatusPolling.tsx

```diff
@@ -39,6 +41,7 @@ function OrderSubmitStatusPolling({
     useEffect(() => {
         if (
             getServiceDetailsByIdQuery.data &&
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
```

Comment:
We must remove this line. 
The error would go if we use     VendorHostedServiceDetailsVo.serviceDeploymentState to compare but the problem is that the "useServiceDetailsPollingQuery" calls two different APIs based on serviceHostingType and both return different data models. 

It would be better if we split it into two different react queries but at the same time not duplicate any code in OrderSubmitStatusPolling. 
Or we can convert it to string and do string value comparison instead of enum .

Reply:
@swaroopar Can we unify the return values ​​of the two interfaces as ServiceDetailsVo? Why define VendorHostedServiceDetailsVo?